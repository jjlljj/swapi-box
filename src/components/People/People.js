import React, { Component } from 'react';
import propTypes from 'prop-types';
import './People.css';
import Card from '../Card/Card';

class People extends Component {
   
  async componentDidMount() {
    if (!this.props.cards) this.props.fetchData()
  }

  renderCards() {
    return this.props.cards.map(card => {
            return (<Card card={card} />)
    })
  }

  render() {
    return (
      <div> 
        { 
          this.props.cards &&
          this.renderCards()
        }
      </div>
    );
  }
};

export default People;
