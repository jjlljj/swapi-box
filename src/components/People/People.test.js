/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import People from './People';

describe('People', () => {

  it('should not pass', () => {
    

    expect(true).toEqual(false)
  })

})
