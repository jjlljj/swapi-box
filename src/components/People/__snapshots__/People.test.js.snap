// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People should pass snapshot when passed a single card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    addToFav={undefined}
    cards={
        Array [
            Object {
              "Homeword": "Tatooine",
              "Population": "200000",
              "Species": "Human",
              "cardType": "people",
              "name": "Beru Whitesun lars",
            },
          ]
    }
    fetchData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addToFav={undefined}
          card={
                    Object {
                              "Homeword": "Tatooine",
                              "Population": "200000",
                              "Species": "Human",
                              "cardType": "people",
                              "name": "Beru Whitesun lars",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Beru Whitesun lars",
        "nodeType": "function",
        "props": Object {
          "addToFav": undefined,
          "card": Object {
            "Homeword": "Tatooine",
            "Population": "200000",
            "Species": "Human",
            "cardType": "people",
            "name": "Beru Whitesun lars",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addToFav={undefined}
            card={
                        Object {
                                    "Homeword": "Tatooine",
                                    "Population": "200000",
                                    "Species": "Human",
                                    "cardType": "people",
                                    "name": "Beru Whitesun lars",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Beru Whitesun lars",
          "nodeType": "function",
          "props": Object {
            "addToFav": undefined,
            "card": Object {
              "Homeword": "Tatooine",
              "Population": "200000",
              "Species": "Human",
              "cardType": "people",
              "name": "Beru Whitesun lars",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`People should pass snapshot when passed cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    addToFav={undefined}
    cards={
        Array [
            Object {
              "Homeword": "Tatooine",
              "Population": "200000",
              "Species": "Droid",
              "cardType": "people",
              "name": "R5-D4",
            },
            Object {
              "Homeword": "Tatooine",
              "Population": "200000",
              "Species": "Human",
              "cardType": "people",
              "name": "Beru Whitesun lars",
            },
          ]
    }
    fetchData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addToFav={undefined}
          card={
                    Object {
                              "Homeword": "Tatooine",
                              "Population": "200000",
                              "Species": "Droid",
                              "cardType": "people",
                              "name": "R5-D4",
                            }
          }
/>,
        <Card
          addToFav={undefined}
          card={
                    Object {
                              "Homeword": "Tatooine",
                              "Population": "200000",
                              "Species": "Human",
                              "cardType": "people",
                              "name": "Beru Whitesun lars",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "R5-D4",
        "nodeType": "function",
        "props": Object {
          "addToFav": undefined,
          "card": Object {
            "Homeword": "Tatooine",
            "Population": "200000",
            "Species": "Droid",
            "cardType": "people",
            "name": "R5-D4",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Beru Whitesun lars",
        "nodeType": "function",
        "props": Object {
          "addToFav": undefined,
          "card": Object {
            "Homeword": "Tatooine",
            "Population": "200000",
            "Species": "Human",
            "cardType": "people",
            "name": "Beru Whitesun lars",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addToFav={undefined}
            card={
                        Object {
                                    "Homeword": "Tatooine",
                                    "Population": "200000",
                                    "Species": "Droid",
                                    "cardType": "people",
                                    "name": "R5-D4",
                                  }
            }
/>,
          <Card
            addToFav={undefined}
            card={
                        Object {
                                    "Homeword": "Tatooine",
                                    "Population": "200000",
                                    "Species": "Human",
                                    "cardType": "people",
                                    "name": "Beru Whitesun lars",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "R5-D4",
          "nodeType": "function",
          "props": Object {
            "addToFav": undefined,
            "card": Object {
              "Homeword": "Tatooine",
              "Population": "200000",
              "Species": "Droid",
              "cardType": "people",
              "name": "R5-D4",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Beru Whitesun lars",
          "nodeType": "function",
          "props": Object {
            "addToFav": undefined,
            "card": Object {
              "Homeword": "Tatooine",
              "Population": "200000",
              "Species": "Human",
              "cardType": "people",
              "name": "Beru Whitesun lars",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
